test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -2067,6 +2067,22 @@ public synchronized Set<String> nodesInclude(String index) {
         return Collections.emptySet();
     }
 
+    public synchronized Set<String> nodesIDsInclude(String index) {
+        if (clusterService().state().routingTable().hasIndex(index)) {
```

Comment:
Given that the metods here get the state multiple times anyway, we might as well call `nodesInclude` and translate the result instead by looking up the name. It could be the new version here that is the foundation, since looking up by `id` is easier (`clusterService().state().getNodes().get(id)`). Or a private method getting the `DiscoveryNode` objects and both map to the right form (id vs name).

I'd prefer that to avoid the duplicate code, thinking that they could drift apart in the future.